src/common/tools/gen_listings.ts

```diff
@@ -5,25 +5,46 @@ import * as process from 'process';
 import { crawl } from './crawl.js';
 
 function usage(rc: number): void {
-  console.error('Usage:');
-  console.error('  tools/gen_listings [OUT_DIR] [SRC_DIR] [SUITES...]');
-  console.error('  tools/gen_listings out/ src/ webgpu unittests');
+  console.error(`Usage: tools/gen_listings [options] [OUT_DIR] [SUITE_DIRS...]
+
+For each suite in SUITE_DIRS, generate listings and write each listing.js
+into OUT_DIR/{suite}/listing.js. Example:
+  tools/gen_listings out/ src/unittests/ src/webgpu/
+
+Options:
+  --help          Print this message and exit.
+  --no-validate   Whether to validate test modules while crawling.
+`);
   process.exit(rc);
 }
 
-if (process.argv.length <= 4) {
+const argv = process.argv;
+if (argv.indexOf('--help') !== -1) {
```

Comment:
FYI we have one other utility with command line flags: `runtime/cmdline.ts`. This looks fine though.